artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -673,15 +738,24 @@ public Page removePage(int pageId) {
                return null;
             }
 
-            Page page = createPage(pageId);
+            Page page = usePage(pageId);
 
-            if (page.getFile().exists()) {
+            if (page != null && page.getFile().exists()) {
+               page.usageDown();
```

Comment:
numberOfPages is orthogonal to usedPages here. it's just keeping the older logic here as I did not change anything related to remove pages.

it could happen a cursor be passing through this and keeping the cache a bit longer.